posthog/api/capture.py

```diff
@@ -236,6 +240,10 @@ def get_event(request):
         if not event.get("properties"):
             event["properties"] = {}
 
+        # Support test_[apiKey] for users with multiple environments
+        if event["properties"].get("test_environment") is None:
+            event["properties"]["test_environment"] = True if is_test_environment else False
```

Comment:
Should probably a property with a `$` to indicate that it's a standard posthog one? Then you can also add it to PropertyKeyInfo.tsx. Also worth adding to `posthog/models/team.py` so it's included in the standard filter.

Reply:
I'll add the `$`. I can it to team/propertykey but should we? Currently, it's meant to be hidden and not selectable in the UI.

Reply:
I've been looking at `posthog/models/team.py` but I'm unable to figure out what you had in mind?

Reply:
re propertykeyinfo, you can `hide: true` to hide it in the event view etc

re: teams, I'd suggest adding a condition to the test filters (https://github.com/PostHog/posthog/blob/master/posthog/models/team.py#L36) so that when someone ticks 'filter test accounts' these events get filtered too.

Reply:
I think there are differences between the way you are thinking about this and the way Paolo and I were thinking about it. I'm struggling to identify exactly what that difference is. I think it has something to do with "filter test accounts" since this feature supersedes that feature. It's not meant to be integrated into the existing filters. It's a matter of the state being managed outside the regular filters. I'll give it some more thought because you may be right about this.

Reply:
So actually @buwilliams, I think we left an asterisk on what to do with the existing test filters. I'm strongly leaning on keeping it as an advanced feature both for backwards-compatibility and support other use cases where changing the event data is not the best user experience.

In this case, the test account filters switch (frontend) would disappear as it is today, and when the new general switch is toggled to test, we would include the internal/test filters too.

From a backend standpoint it might actually be a good way to go to rely on the existing test filters logic to filter out these events instead of adding the `test_environment` property to each request (as Tim is suggesting). Thoughts?

Reply:
@paolodamico adding the `test_environment` property may be necessary since the current "filter test data" assumes the data needed to filter is present. Otherwise, how do you filter for test environments if we currently are not tracking that an environment is a test environment? Additionally, the case was for non-technical users who by adding `test_` to their apiKey to get this functionality for free. Adding test_[apiKey] allows us to inject that information so that the filtering test data has a property to use. I think that is a nice user experience otherwise users will need to manually add `test_environment` to all their capture calls. URLs won't work since we support more than browsers.

re: "advanced feature both for backwards-compatibility and support other use cases where changing the event data is not the best user experience" - What is an advanced feature? What makes an advanced feature backwards compatible? Why does change event data not a good user experience? I could see only adding the test_environment in the test_ case and not all cases but I'm unsure if that is what you mean.

re: exist test filters logic - Leveraging existing test filters logic seems good to me.